src/centroiders.cpp

```diff
@@ -1,95 +1,199 @@
-    return sqrt(distX*distX + distY*distY);
-}
-
-// helper for StarCentroidsCompare
-static std::vector<int> FindClosestCentroids(float threshold,
-                                              std::vector<Star> one,
-                                              std::vector<Star> two) {
-    std::vector<int> result;
-
-    for (int i = 0; i < (int)one.size(); i++) {
-        float closestDistance = INFINITY;
-        int   closestIndex = -1;
-
-        for (int k = 0; k < (int)two.size(); k++) {
-            float currDistance = StarDistancePixels(one[i], two[k]);
-            if (currDistance < threshold && currDistance < closestDistance) {
-                closestDistance = currDistance;
-                closestIndex = k;
-            }
-        }
-        
-        result.push_back(closestIndex);
-    }
-
-    return result;
-}
-
-CentroidComparison StarCentroidsCompare(float threshold,
-                                    std::vector<Star> expected,
-                                    std::vector<Star> actual) {
-
-    CentroidComparison result;
-    // maps from indexes in each list to the closest centroid from other list
-    std::vector<int> expectedToActual, actualToExpected;
-
-    expectedToActual = FindClosestCentroids(threshold, expected, actual);
-    actualToExpected = FindClosestCentroids(threshold, actual, expected);
-
-    // any expected stars whose closest actual star does not refer to them are missing
-    for (int i = 0; i < (int)expectedToActual.size(); i++) {
-        if (expectedToActual[i] == -1 ||
-            actualToExpected[expectedToActual[i]] != i) {
-            result.numMissingStars++;
-        } else {
-            result.meanError += StarDistancePixels(expected[i], actual[expectedToActual[i]]);
-        }
-    }
-    result.meanError /= (expected.size() - result.numMissingStars);
-
-
-    // any actual star whose closest expected star does not refer to them is extra
-    for (int i = 0; i < (int)actual.size(); i++) {
-        if (actualToExpected[i] == -1 ||
-            expectedToActual[actualToExpected[i]] != i) {
-            result.numExtraStars++;
-        }
-    }
-
-    return result;
-}
-
-void CentroidComparison::Print() {
-    printf("Extra stars: %d\nMissing stars: %d\nMean error: %f\n",
-           numExtraStars, numMissingStars, meanError);
-}
-
-}
+#include "centroiders.hpp"
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <math.h>
+
+#include <vector>
+#include <iostream>
+
+#include <unordered_map>
+
+#include <unordered_set>
+
+namespace lost {
+
+// DUMMY
+
+std::vector<Star> DummyCentroidAlgorithm::Go(unsigned char *image, int imageWidth, int imageHeight) const {
+    std::vector<Star> result;
+
+    for (int i = 0; i < numStars; i++) {
+        result.push_back(Star(rand() % imageWidth, rand() % imageHeight, 10.0));
+    }
+
+    return result;
+}
+
+//recursive helper here
+void cogHelper(float &yCoordMagSum,  
+    float &xCoordMagSum, 
+    int &magSum, 

```

Comment:
There are so many parameters! It will be better to create a new struct or class that contains all the "state": at least the magsums, the maxs, and the mins. Then rather than passing around a half dozen parameters to each recursion call, you just pass a single state object.

You don't need to declare this in the hpp file, since the class will only be used internally.

Reply:
Done - defined params as a struct and just called the struct inside of Go and Helper

Reply:
We can also expand this to house the many params that we will surely need for the other algorithms. Just call it in those as well in order to have everything in one struct.